config.go

```diff
@@ -0,0 +1,223 @@
+		for key, value := range marshalableConfig.BGColors {
+			bgColors[priorityMapFromString[key]] = value
+		}
+		config.BGColors = bgColors
+	}
+}
+
+func (configIOImpl ConfigIOImpl) Deserialize(reader io.Reader) (marshalableConfig *MarshalableConfig, err error) {
+	decoder := json.NewDecoder(reader)
+	marshalableConfig = &MarshalableConfig{}
+	err = decoder.Decode(marshalableConfig)
+	return marshalableConfig, err
+}
+
+func loadConfigurationFile(config *config) (marshalableConfig *MarshalableConfig, err error) {
+	if file, err := os.Open(config.ConfigFile); err == nil {
+		defer file.Close()
+		configIO := NewConfigIO()
+		return configIO.Deserialize(file)
+	}
+	return nil, err
+}
+
+func copyToConfigFromCMDOptions(config *config) {
+	kingpin.Flag("priority", "Priority of newly created tasks (veryhigh,high,medium,low,verylow).").Short('p').EnumVar(&config.Priority, "veryhigh", "high", "medium", "low", "verylow")
+	kingpin.Flag("graft", "Task to graft new tasks to.").Short('g').Default("root").StringVar(&config.Graft)
+	kingpin.Flag("file", "File to load task lists from.").StringVar(&config.File)
+	kingpin.Flag("legacy-file", "File to load legacy task lists from.").StringVar(&config.LegacyFile)
+	kingpin.Flag("order", "Specify display order of tasks (index,created,completed,text,priority,duration,done).").EnumVar(&config.Order, "index", "created", "completed", "text", "priority", "duration", "done")
+
+	colors := []string{WHITE, BLUE, RED, CYAN, GREEN, YELLOW, BLACK, MAGENTA, BRIGHTWHITE, BRIGHTBLUE, BRIGHTRED, BRIGHTCYAN, BRIGHTGREEN, BRIGHTYELLOW, BRIGHTBLACK, BRIGHTMAGENTA, NOCOLOR}
+
+	var veryLowFGColor = kingpin.Flag("verylowfgcolor", "Very low priority task texts foreground color.").Default(NOCOLOR).Enum(colors...)
+	var lowFGColor = kingpin.Flag("lowfgcolor", "Low priority task texts foreground color.").Default(NOCOLOR).Enum(colors...)
+	var mediumFGColor = kingpin.Flag("mediumfgcolor", "Medium priority task texts foreground color.").Default(NOCOLOR).Enum(colors...)
+	var highFGColor = kingpin.Flag("highfgcolor", "High priority task texts foreground color.").Default(NOCOLOR).Enum(colors...)
+	var veryHighFGColor = kingpin.Flag("veryhighfgcolor", "Very high task texts foreground color.").Default(NOCOLOR).Enum(colors...)
+	var veryLowBGColor = kingpin.Flag("verylowbgcolor", "Very low priority task texts background color.").Default(NOCOLOR).Enum(colors...)
+	var lowBGColor = kingpin.Flag("lowbgcolor", "Low priority task texts background color.").Default(NOCOLOR).Enum(colors...)
+	var mediumBGColor = kingpin.Flag("mediumbgcolor", "Medium priority task texts background color.").Default(NOCOLOR).Enum(colors...)
+	var highBGColor = kingpin.Flag("highbgcolor", "High priority task texts background color.").Default(NOCOLOR).Enum(colors...)
+	var veryHighBGColor = kingpin.Flag("veryhighbgcolor", "Very high priority task texts background color.").Default(NOCOLOR).Enum(colors...)
+	kingpin.Parse()
+
+	if *veryLowFGColor != NOCOLOR {
```

Comment:
Because i don't want these flag defaults to override the configuration loaded from the configuration file prior to this. I do not think the 
`.Default(NOCOLOR)`
is that pretty though. I was thinking about removing those defaults anyway, and just check if the variables are set or not.